README.md

```diff
@@ -72,3 +80,55 @@ The following make targets are provided:
 The generated Thrift code is committed to the Git repo, so if you change
 `edgecontext.thrift` make sure to run `make thrift fmt` and commit those
 changes as well.
+
+## Go
+
+### Usage
+
+Use [`edgecontext.Factory`] to create an [`ecinterface.Factory`] implementation
+that's expected by [`baseplate.New`]:
+
+```go
+ctx, bp, err := baseplate.New(context.Background(), baseplate.NewArgs{
+  ConfigPath: configPath,
+  ServiceCfg: &cfg, // or nil if you don't have additional config to parse
+  EdgeContextFactory: edgecontext.Factory(edgecontext.Config{
+    Logger: log.ErrorWithSentryWrapper(),
+  }),
+})
+```
+
+When using it, get the [`*EdgeRequestContext`] object out of context:
+
+```go
+if ec, ok := edgecontext.GetEdgeContext(ctx); ok {
+  user := ec.User()
+  loid, ok := user.LoID()
+  // Do something with loid
+}
+```
+
+[`edgecontext.Factory`]: https://pkg.go.dev/github.com/reddit/edgecontext/lib/go/edgecontext#Factory
+[`ecinterface.Factory`]: https://pkg.go.dev/github.com/reddit/baseplate.go/ecinterface#Factory
+[`baseplate.New`]: https://pkg.go.dev/github.com/reddit/baseplate.go#New
+[`*EdgeRequestContext`]: https://pkg.go.dev/github.com/reddit/edgecontext/lib/go/edgecontext#EdgeRequestContext
+
+### Development
```

Comment:
I can add an `all-thrift` rule in the top level directory. The problem is that the python part run fmt on the thrift generated code, and fmt can only be run from inside docker. Which means after running `all-thrift` we still need to run the additional `py-fmt` from inside docker. of course this `all-thrift` rule can be run inside docker. but it will also be a hassle to prepare a docker image with both py and go (we do not need that for other go developments, but we would need that for `all-thrift` if that's the only thrift rule).

go developments do not require docker, everything can be done locally.

alternatively we could pioneer poetry for edgecontext py, so instead of preparing a docker image, we just use a venv managed by poetry to run those py make targets. `all-thrift` can also be run from that venv instead.

Reply:
I understand that you don't need docker in the general case for go, but this is a shared library that a lot of people without fully configured environments will end up using so if we can just get a full dev env into one docker container and use that at the root for everything it'll make it easier for everyone.